grep.json

```diff
@@ -0,0 +1,231 @@
+{ 
+    "#": [
+        "JSON doesn't allow for multi-line strings, so all outputs are presented ",
+        "here as arrays of strings. It's up to the test generator to join the ", 
+        "lines together with line breaks.",
+        "",
+        "The tests are divided into two groups: ",
+        "* Grepping a single file",
+        "* Grepping multiple files at once",
+        "",
+        "The language track implementing this exercise should",
+        "ensure that when the tests run, three files are created",
+        "with the following contents:",
+        "",
+        " [File #1]",
+        " File name: iliad.txt ",
+        " Contents: ",
+        "Achilles sing, O Goddess! Peleus' son;",
+        "His wrath pernicious, who ten thousand woes",
+        "Caused to Achaia's host, sent many a soul",
+        "Illustrious into Ades premature,",
+        "And Heroes gave (so stood the will of Jove)",
+        "To dogs and to all ravening fowls a prey,",
+        "When fierce dispute had separated once",
+        "The noble Chief Achilles from the son",
+        "Of Atreus, Agamemnon, King of men.",
+        "",
+        " [File #2] ",
+        " File name: midsummer-night.txt ",
+        " Contents: ",
+        "I do entreat your grace to pardon me.",
+        "I know not by what power I am made bold,",
+        "Nor how it may concern my modesty,",
+        "In such a presence here to plead my thoughts;",
+        "But I beseech your grace that I may know",
+        "The worst that may befall me in this case,",
+        "If I refuse to wed Demetrius.",
+        "",
+        " [File #3] ",
+        " File name: paradise-lost.txt ",
+        " Contents: ",
+        "Of Mans First Disobedience, and the Fruit",
+        "Of that Forbidden Tree, whose mortal tast",
+        "Brought Death into the World, and all our woe,",
+        "With loss of Eden, till one greater Man",
+        "Restore us, and regain the blissful Seat,",
+        "Sing Heav'nly Muse, that on the secret top",
+        "Of Oreb, or of Sinai, didst inspire",
+        "That Shepherd, who first taught the chosen Seed"
+    ],
+    "single-file": {
+        "description": ["Test grepping a single file"],
+        "cases": [
+            {
+                "description": "One file, one match, no flags",
+                "pattern": "Agamemnon",
+                "flags": "",
+                "files": ["iliad.txt"],
+                "expected": "Of Atreus, Agamemnon, King of men."
```

Comment:
The fact that some `expected` are strings and some are arrays may mess up languyages which expect a fixed type for `expected` - I would have to ee whether it's possible. The issue can be avoided by always using array, even if there is only one element

(I know that the arrays have to be joined back together to be a string anyway, but the above comment is for the benefit of tracks that wish to automatically generate their tests by reading this JSON file)